.github/workflows/build+upload.yml

```diff
@@ -0,0 +1,52 @@
+name: Build and upload
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag-override:
+        description: "Tag"
+        type: string
+        required: false
+
+env:
+  java-version: "17"
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Apply tag override and parse ref"
+        uses: bisgardo/github-action-parse-ref@v1
+        id: ref
+        with:
+          ref: "${{github.event.inputs.tag-override || github.ref}}"
+          default-ref-type: tags
+      - name: Fail if ref is not a tag
```

Comment:
We should also check that the tag version matches the one specified in `build.gradle`.

Reply:
Deferring this to a separate PR to unblock `build+test` which currently gets cancelled due to resource consumption.